File: Data Structures & Algorithms/sliding-window-maximum/submission-4.cpp
class Solution {
public:
    vector<int> maxSlidingWindow(vector<int>& nums, int k) {
            int l=0,r=0;
            int Max = INT_MIN;
            vector<int> res;    
            while(r<nums.size()){
                Max = max(Max,nums[r]);
             
                if((r-l+1)==k){
                    // cout<<true<<endl;
                    res.push_back(Max);
                    Max=INT_MIN;
                    r=l;
                    l+=1;

                }
                r++;
            }
   

            return res;
   
   
    }  
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-4.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    longestConsecutive(nums) {
        let maxLen = 0;
        let seen = new Set(nums);
        
        for(let i=0;i<nums.length;i++){
           if(!seen.has(nums[i]-1)){
                let len = 1;
                let curr = nums[i];

                while(seen.has(curr+1)){
                    curr++;
                    len++;
                }
                maxLen = Math.max(len,maxLen);
           }
        }
        return maxLen;
    }
}
